src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -341,24 +378,55 @@ export class SavedObjectsRepository {
-          client: this.client,
-          serializer: this._serializer,
-          getIndexForType: this.getIndexForType.bind(this),
-          createPointInTimeFinder: this.createPointInTimeFinder.bind(this),
-          objects: [{ type, id, overwrite, namespaces: initialNamespaces ?? [namespaceString] }],
-        });
-        if (error) {
-          throw SavedObjectsErrorHelpers.createConflictError(type, id);
-        }
-        savedObjectNamespaces =
-          initialNamespaces || getSavedObjectNamespaces(namespace, existingDocument);
-        existingOriginId = existingDocument?._source?.originId;
-      } else {
-        savedObjectNamespaces = initialNamespaces || getSavedObjectNamespaces(namespace);
+        preflightResult = (
+          await preflightCheckForCreate({
+            registry: this._registry,
+            client: this.client,
+            serializer: this._serializer,
+            getIndexForType: this.getIndexForType.bind(this),
+            createPointInTimeFinder: this.createPointInTimeFinder.bind(this),
+            objects: [{ type, id, overwrite, namespaces: initialNamespaces ?? [namespaceString] }],
+          })
+        )[0];
       }
+      savedObjectNamespaces =
+        initialNamespaces || getSavedObjectNamespaces(namespace, preflightResult?.existingDocument);
+      existingOriginId = preflightResult?.existingDocument?._source?.originId;
+    }
+
+    const spacesToEnforce = new Set(initialNamespaces).add(namespaceString); // Always check/enforce authZ for the active space
+    const existingNamespaces = preflightResult?.existingDocument?._source?.namespaces || [];
+    const spacesToAuthorize = new Set(existingNamespaces);
+    spacesToAuthorize.delete(ALL_NAMESPACES_STRING); // Don't accidentally check for global privileges when the object exists in '*'
+    const authorizationResult = await this._securityExtension?.checkAuthorization({
+      types: new Set([type]),
+      spaces: new Set([...spacesToEnforce, ...spacesToAuthorize]), // existing namespaces are included so we can later redact if necessary
+      actions: ['create'],
+      // If a user tries to create an object with `initialNamespaces: ['*']`, they need to have 'create' privileges for the Global Resource
+      // (e.g., All privileges for All Spaces).
+      // Inversely, if a user tries to overwrite an object that already exists in '*', they don't need to 'create' privileges for the Global
+      // Resource, so in that case we have to filter out that string from spacesToAuthorize (because `allowGlobalResource: true` is used
+      // below.)
+      options: { allowGlobalResource: true },
+    });
+    if (authorizationResult) {
+      this._securityExtension!.enforceAuthorization({
+        typesAndSpaces: new Map([[type, spacesToEnforce]]),
+        action: 'create',
+        typeMap: authorizationResult.typeMap,
+        auditCallback: (error) =>
+          this._securityExtension!.addAuditEvent({
```

Comment:
I can't find any tests for any of the `addAuditEvent` calls in `repository.test.ts`. 

Are they located elsewhere?

Reply:
@thomheymann, Jeramy split out the security related tests (including auditing) into its own suite: `src/core/server/saved_objects/service/lib/repository.security_extension.test.ts`.

You should see audit tests there similar to the following: https://github.com/jportner/kibana/blob/d8110c4b9e858d2a1996a43e13fe99158e21b486/src/core/server/saved_objects/service/lib/repository.security_extension.test.ts#L234-L262